### PR TITLE
Refactor estimate_KM

### PR DESCRIPTION
**Pull request by rebecca-albrecht**
I refactored the estimate_KM function as discussed in the core meeting (making the implicit use of CDISC ADaM explicit) and adjusted the tests accordingly.

### CODE REVIEW

**Comment by rebecca-albrecht** [comment]
@cschaerfe @SHAESEN2 The functionality that columns PARAM/PARAMCD are treated like strata (if strata is not specified) is still om there. Ok?

File: R/estimate_KM.R [hunk 1]
```diff
@@ -1,8 +1,7 @@
-#' @title Wrapper for Kaplan Meier analysis for an ADaM Basic Data Structure (BDS) for Time-to-Event analysis
+#' @title Wrapper for Kaplan Meier Time-to-Event analysis
```
Comment (SHAESEN2) [inline R/estimate_KM.R:1]: I would keep the references to CDISC standards as this is a big advantage that we can do this.

File: R/estimate_KM.R [hunk 2]
```diff
@@ -1,8 +1,7 @@
...
 #'  
-#' @description This function is a wrapper around \code{survival::survfit.formula} to perform a Kaplan-Meier analysis,
-#'    based on the expected ADaM Basic Data Structure (BDS) for Time-to-Event analysis and assuming right-censored data.
-#'    The function expects that the data has been filtered on the PARAM/PARAMCD of interest.
+#' @description This function is a wrapper around \code{survival::survfit.formula} to perform a Kaplan-Meier analysis, assuming right-censored data.
```
Comment (SHAESEN2) [inline R/estimate_KM.R:3]: keep reference to CDISC, agree with right-censored data for now.

File: R/estimate_KM.R [hunk 3]
```diff
@@ -17,9 +16,11 @@
 #' 
 #' @seealso \code{\link[survival]{survfit.formula} \link[survival]{survfitCI}}
 #' 
-#' @param data The name of the ADaM Basic Data Structure (BDS) for Time-to-Event analysis eg ADTTE. Rows in which AVAL or CNSR contain NA, are removed during analysis. 
-#' @param strata Character vector, representing the strata for Time-to-Event analysis eg TRT01P. When NULL, an overall analysis is performed.
+#' @param data Data.frame for Time-to-Event analysis. Rows in which the analysis variable (AVAL) or the sensor variable (CNSR) contain NA, are removed during analysis.
```
Comment (SHAESEN2) [inline R/estimate_KM.R:19]: + include reference to CDISC

File: R/estimate_KM.R [hunk 4]
```diff
@@ -17,9 +16,11 @@
 #' 
 #' @seealso \code{\link[survival]{survfit.formula} \link[survival]{survfitCI}}
 #' 
-#' @param data The name of the ADaM Basic Data Structure (BDS) for Time-to-Event analysis eg ADTTE. Rows in which AVAL or CNSR contain NA, are removed during analysis. 
-#' @param strata Character vector, representing the strata for Time-to-Event analysis eg TRT01P. When NULL, an overall analysis is performed.
+#' @param data Data.frame for Time-to-Event analysis. Rows in which the analysis variable (AVAL) or the sensor variable (CNSR) contain NA, are removed during analysis. 
+#' @param strata Character vector, representing the strata for Time-to-Event analysis. When NULL, an overall analysis is performed.
 #'   Default is NULL.
+#' @param AVAL Analysis value for Time-to-Event analysis. Default is "AVAL".
```
Comment (SHAESEN2) [inline R/estimate_KM.R:22]: Do we need to explain that CNSR = 1 is assumed censored?

File: R/estimate_KM.R [hunk 5]
```diff
@@ -65,13 +68,14 @@ estimate_KM <- function(
...
  ### map passes .x as Call$data
  ### df: catch expressions that represent base R subsets
-  
   Call <- as.list(match.call())
   dots <- list(...)
   dfExpr <- Call[["data"]]
   
  ## Validate `data` and capture data name
  
+  reqcols <- c(strata, CNSR, AVAL)
```
Comment (SHAESEN2) [inline R/estimate_KM.R:77]: move back to validate columns section

**Review by SHAESEN2 — COMMENTED** [review]
Hey Rebecca,

Please keep references to CDISC as we want to attract the attention of pharma as well. 
Nice work !
Make sure to run the adjusted tests and see that they don't fail.
Did I miss the update in the test for the survival::survfit ... equality?

File: R/estimate_KM.R (hunk 5 above)
Comment (rebecca-albrecht) [reply R/estimate_KM.R:77]: How do you mean, move back?

File: R/estimate_KM.R (hunk 1 above)
Comment (rebecca-albrecht) [reply R/estimate_KM.R:1]: I had the feeling that some core members wanted it to not be as present as it was (@bailliem?). But I might be wrong. I can leave it in for sure.

File: R/estimate_KM.R (hunk 5 above)
Comment (SHAESEN2) [reply R/estimate_KM.R:77]: This part of the code was under the next point ## validate columns

Comment (rebecca-albrecht) [reply R/estimate_KM.R:77]: Ah, thank you!

**Comment by cschaerfe** [comment]
@SHAESEN2 In regards to the CDISC reference, I suggest to keep it out of the function doc, but maybe provide a CDISC specific vignette showing how the functions are to be used with CDISC Adam

File: R/estimate_KM.R (hunk 4 above)
Comment (cschaerfe) [reply R/estimate_KM.R:22]: Yes, I think that would be good

**Pull request merged**